DegreeProject/Assets/Scripts/Enemy/AI.cs

```diff
@@ -116,23 +124,22 @@ private IEnumerator HurtAnimCd(float seconds)
         hurtCd = false;
     }
 
-    private IEnumerator HurtAnimation(float delay)
+    private IEnumerator Die(float delay)
     {
+        if (gameObject == null)
+            yield break;
+
+        animator.StopPlayback(); // Stop current animation?
+        animator.SetTrigger(StringData.death);
         yield return new WaitForSeconds(delay);
-        animator.SetTrigger(StringData.hurt);
+        Spawner.SpawnWhenDied(1);
+        RestoreHealth(gameObject);
+        ObjectPool.Despawn(gameObject);
     }
 
-    private IEnumerator Die(float delay)
+    private void HurtAnimation()
     {
-        if (gameObject != null)
-        {
-            animator.StopPlayback(); // Stop current animation?
-            animator.SetTrigger(StringData.death);
-            yield return new WaitForSeconds(delay);
-            Spawner.SpawnWhenDied(1);
-            RestoreHealth(gameObject);
-            ObjectPool.Despawn(gameObject);
-        }
+        animator.SetTrigger(StringData.hurt);
```

Comment:
Du kan vilja kolla om du just nu spelar Hurt-animationen. Triggern sätts till off när animatorn går in i animationen, vilket gör att du kan få din hurt animation trigger att bli on igen medans hurt spelas. Detta kan göra att du får en hurt animation spelandes när du inte vill det senare, då triggern är satt till on tills animatorn gått in i animationen igen.